Assets/Mirror/Runtime/NetworkIdentity.cs

```diff
@@ -147,6 +147,21 @@ internal set
 
         public NetworkBehaviour[] NetworkBehaviours => networkBehavioursCache = networkBehavioursCache ?? GetComponents<NetworkBehaviour>();
 
+
+        // NetworkProximityChecker caching
+        NetworkProximityCheck proximityCheckCache;
```

Comment:
Is ProximityCheck the only way an object can be visable or hidden ? 

If there are more ways then we'd need a way to combine different checks

Reply:
it's the only way.
if there is a prox checker then we use it to check for observers.
if there is none then we add all server connections.